### PR TITLE
Add support for logout wreply validation in WS-Federation (Passive) Configuration

### PR DESCRIPTION
**Pull request by Lakshan-Banneheke**
### Proposed changes in this pull request

- Add field to WS-Federation (Passive) Configuration in an SP to configure wreply logout url.
Screenshot 2023-11-02 at 2 15 41 PM

- Add config to identity.xml to enable/disable logout wreply validation.

### When should this PR be merged
No preconditions

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 1]
```diff
@@ -917,6 +920,40 @@ public String getPassiveSTSWReply() {
...
+        if (passiveSTSWReplyLogout != null) {
+            return passiveSTSWReplyLogout;
+        }
+
+        InboundAuthenticationRequestConfig[] authRequest = serviceProvider
+                .getInboundAuthenticationConfig().getInboundAuthenticationRequestConfigs();
+
+        if (authRequest != null) {
+            for (int i = 0; i < authRequest.length; i++) {
+                if ("passivests".equalsIgnoreCase(authRequest[i].getInboundAuthType())) {
```
Comment (ShanChathusanda93) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:939]: Let's move this to a constant.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 2]
```diff
@@ -1631,6 +1679,15 @@ public void setPassiveSTSWReply(String passiveSTSWReply) {
         this.passiveSTSWReply = passiveSTSWReply;
     }
 
+    /**
+     * Set passive STS wreply logout url.
+     *
+     * @param passiveSTSWReplyLogout Wreply logout url.
+     */
+    public void setPassiveSTSWReplyLogout(String passiveSTSWReplyLogout) {
+        this.passiveSTSWReplyLogout = passiveSTSWReplyLogout;
```
Comment (ShanChathusanda93) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:1688]: Formatting

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java (hunk 1 above)
Comment (Lakshan-Banneheke) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java]: Fixed with [ab85885](https://github.com/wso2/carbon-identity-framework/pull/5129/commits/ab8588512be9c241a44f76890e44d705a568b6ef)

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java (hunk 2 above)
Comment (Lakshan-Banneheke) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:1690]: Fixed with [ab85885](https://github.com/wso2/carbon-identity-framework/pull/5129/commits/ab8588512be9c241a44f76890e44d705a568b6ef)

File: components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/IdentityConstants.java [hunk 3]
```diff
@@ -410,6 +410,8 @@ public static class STS {
                 "PassiveSTS.DisableAppliesToInPassiveSTSResponse";
         public static final String PASSIVE_STS_ENABLE_DEFAULT_SIGNATURE_AND_DIGEST_ALG =
                 "PassiveSTS.EnableDefaultSignatureAndDigestAlgorithm";
+        public static final String PASSIVE_STS_LOGOUT_WREPLY_VALIDATION =
+                "PassiveSTS.LogoutWreplyValidation";
```
Comment (madurangasiriwardena) [inline components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/IdentityConstants.java:414]:
```suggestion
                "PassiveSTS.EnableLogoutWreplyValidation";
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 4]
```diff
@@ -829,6 +829,7 @@
         <SLOHostNameVerificationEnabled>true</SLOHostNameVerificationEnabled>
         <DisableAppliesToInPassiveSTSResponse>false</DisableAppliesToInPassiveSTSResponse>
         <EnableDefaultSignatureAndDigestAlgorithm>true</EnableDefaultSignatureAndDigestAlgorithm>
+        <LogoutWreplyValidation>true</LogoutWreplyValidation>
```
Comment (madurangasiriwardena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:832]:
```suggestion
        <EnableLogoutWreplyValidation>true</EnableLogoutWreplyValidation>
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 5]
```diff
@@ -1220,6 +1220,7 @@
         <SLOHostNameVerificationEnabled>{{passive_sts.slo.host_name_verification}}</SLOHostNameVerificationEnabled>
         <DisableAppliesToInPassiveSTSResponse>{{passive_sts.disable_applies_to_in_response}}</DisableAppliesToInPassiveSTSResponse>
         <EnableDefaultSignatureAndDigestAlgorithm>{{passive_sts.enable_default_signature_and_digest_alg}}</EnableDefaultSignatureAndDigestAlgorithm>
+        <LogoutWreplyValidation>{{passive_sts.logout_wreply_validation}}</LogoutWreplyValidation>
```
Comment (madurangasiriwardena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1223]:
```suggestion
        <LogoutWreplyValidation>{{passive_sts.enable_logout_wreply_validation}}</LogoutWreplyValidation>
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 6]
```diff
@@ -1330,18 +1368,29 @@ public void update(HttpServletRequest request) {
...
         String passiveSTSRealm = request.getParameter("passiveSTSRealm");
         String passiveSTSWReply = request.getParameter("passiveSTSWReply");
+        String passiveSTSWReplyLogout = request.getParameter(PASSIVE_STS_REPLY_URL_LOGOUT);
 
         if (StringUtils.isNotBlank(passiveSTSRealm)) {
             InboundAuthenticationRequestConfig opicAuthenticationRequest = new InboundAuthenticationRequestConfig();
             opicAuthenticationRequest.setInboundAuthKey(passiveSTSRealm);
             opicAuthenticationRequest.setInboundAuthType("passivests");
+
+            List<Property> propertiesArrList = new ArrayList<Property>();
```
Comment (madurangasiriwardena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:1378]:
```suggestion
            List<Property> propertiesArrList = new ArrayList<>();
```

File: components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/IdentityConstants.java (hunk 3 above)
Comment (Lakshan-Banneheke) [reply components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/IdentityConstants.java]: Updated with [9eb8e55](https://github.com/wso2/carbon-identity-framework/pull/5129/commits/9eb8e5504074958b8743561daffea34c40478234)

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml (hunk 4 above)
Comment (Lakshan-Banneheke) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Updated with [9eb8e55](https://github.com/wso2/carbon-identity-framework/pull/5129/commits/9eb8e5504074958b8743561daffea34c40478234)

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java (hunk 6 above)
Comment (Lakshan-Banneheke) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java]: Updated with [9eb8e55](https://github.com/wso2/carbon-identity-framework/pull/5129/commits/9eb8e5504074958b8743561daffea34c40478234)

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 5 above)
Comment (Lakshan-Banneheke) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Updated with [9eb8e55](https://github.com/wso2/carbon-identity-framework/pull/5129/commits/9eb8e5504074958b8743561daffea34c40478234)

**Review by madurangasiriwardena — APPROVED** [review]

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 7]
```diff
@@ -917,6 +921,40 @@ public String getPassiveSTSWReply() {
...
+    public String getPassiveSTSWReplyLogout() {
+
+        if (passiveSTSWReplyLogout != null) {
+            return passiveSTSWReplyLogout;
+        }
+
+        InboundAuthenticationRequestConfig[] authRequest = serviceProvider
+                .getInboundAuthenticationConfig().getInboundAuthenticationRequestConfigs();
+
+        if (authRequest != null) {
```
Comment (ayeshajay) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:938]: Shall we use stringutils in not blank for this? please check other places as well

Comment (Lakshan-Banneheke) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:938]: Cannot use StringUtils.isNotBlank() as authRequest object is not a String. Only Strings can be used with StringUtils.isNotBlank().

**Pull request merged**